hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -278,20 +311,134 @@ std::vector<hardware_interface::CommandInterface> GenericSystem::export_command_
   return command_interfaces;
 }
 
+return_type GenericSystem::prepare_command_mode_switch(
+  const std::vector<std::string> & start_interfaces,
+  const std::vector<std::string> & stop_interfaces)
+{
+  hardware_interface::return_type ret_val = hardware_interface::return_type::OK;
+
+  start_modes_.clear();
+  stop_modes_.clear();
+
+  // Starting interfaces
+  // add start interface per joint in tmp var for later check
+  for (const auto & key : start_interfaces)
+  {
+    for (auto i = 0u; i < info_.joints.size(); i++)
+    {
+      if (key == info_.joints[i].name + "/" + hardware_interface::HW_IF_POSITION)
+      {
+        start_modes_.push_back(hardware_interface::HW_IF_POSITION);
+      }
+      if (key == info_.joints[i].name + "/" + hardware_interface::HW_IF_VELOCITY)
+      {
+        start_modes_.push_back(hardware_interface::HW_IF_VELOCITY);
+      }
+    }
```

Comment:
```suggestion
  hardware_interface::return_type ret_val = hardware_interface::return_type::OK;

  // Currently we don't accept ACCELERATION-only and EFFORT-only interfaces
  for (const auto & key : start_interfaces)
  {
    for (auto i = 0ul; i < info_.joints.size(); i++)
    {
      if (key == info_.joints[i].name + "/" + hardware_interface::HW_IF_ACCELERATION || 
          key == info_.joints[i].name + "/" + hardware_interface::HW_IF_EFFORT)
      {
        ret_val = false;
      }
    }
    
    return ret_val;
```

Reply:
Also not allowed if velocity is missing and there are only ACC and POS interfaces.